apps/mobile/src/components/wallet-settings/remove-wallet-modal.tsx

```diff
@@ -1,33 +1,53 @@
 import { RefObject } from 'react';
 
+import { useSettings } from '@/store/settings/settings.write';
 import { t } from '@lingui/macro';
 
-import { SheetRef } from '@leather.io/ui/native';
-
-import { WarningSheet } from '../sheets/warning-sheet';
+import {
+  Box,
+  Button,
+  ErrorTriangleIcon,
+  Sheet,
+  SheetHeader,
+  SheetRef,
+  Text,
+} from '@leather.io/ui/native';
 
 interface RemoveWalletModalProps {
   sheetRef: RefObject<SheetRef>;
   onSubmit(): unknown;
 }
 export function RemoveWalletModal({ sheetRef, onSubmit }: RemoveWalletModalProps) {
+  const { theme: themeVariant } = useSettings();
   return (
-    <WarningSheet
-      sheetRef={sheetRef}
-      title={t`Remove wallet`}
-      variant="critical"
-      description={t`The wallet will be removed from this device. You will lose access to all tokens and collectibles associated with this wallet. 
-                     Before proceeding, make sure you have securely saved your secret key. Without it, you won't be able to access your tokens or collectibles from another device.`}
-      submitButton={{
-        title: t`Proceed`,
-        onPress: onSubmit,
-      }}
-      skipButton={{
-        title: t`Cancel`,
-      }}
-      onPressSupport={() => {
-        // TODO: add support link
-      }}
-    />
+    <Sheet ref={sheetRef} themeVariant={themeVariant}>
+      <Box p="5" justifyContent="space-between" gap="5">
+        <Box gap="5">
+          <Box>
+            <SheetHeader
+              title={t`Remove wallet`}
+              icon={
+                <Box bg="red.background-primary" borderRadius="round" flexDirection="row" p="2">
+                  <ErrorTriangleIcon color="red.action-primary-default" />
+                </Box>
+              }
+              onPressSupport={() => {
+                // TODO: add support link
+              }}
+            />
+          </Box>
+          <Text>{t`The wallet will be removed from this device. You will lose access to all tokens and collectibles associated with this wallet. 
+                     Before proceeding, make sure you have securely saved your secret key. Without it, you won't be able to access your tokens or collectibles from another device.`}</Text>
+        </Box>
+        <Box gap="3">
+          <Button onPress={onSubmit} buttonState="default" title={t`Proceed`} />
+          <Button
+            onPress={() => sheetRef.current?.dismiss()}
+            buttonState="ghost"
+            title={t`Cancel`}
+          />
+        </Box>
+      </Box>
+    </Sheet>
```

Comment:
Previously this component was a really clean wrapper around a layout-like component. Any chance we can continue to separate logic and styles per [layout convention](https://www.notion.so/trustmachines/Use-layout-components-b94ad409bbf5417395a2bc9d22574da1?pvs=4)

Reply:
I can refactor it again and try clean it up. 

I changed it as the original was accepting `variant="critical"` that fed into lookup functions to determine icon colour. 

I can refactor but this is actually a layout component with no real logic , just avoiding a wrapper layer to drill props down to.

Reply:
I reverted this but with a cleaner version of `WarningSheet` without using lookup functions for colours. 

I'm not sure about this as we have several layers of wrapper components in different folders. Heres an example:

- `wallet-settings/configure/[wallet]/index.tsx` imports `<RemoveWalletSheet` from `src/components/wallet-settings/remove-wallet-sheet`
- `<RemoveWalletSheet` is a wrapper component that loads another wrapper - `WarningSheet` from `src/components/sheets`
- `WarningSheet` is in itself a layout component configuring `Sheet` from the UI library. 

What this achieves is passing a different icon / icon colour and some text. 

I think it would be cleaner to extend the code of `Sheet` and then load that directly without all of the layers and drilling.